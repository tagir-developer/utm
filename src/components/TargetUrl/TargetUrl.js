import React, { Fragment } from 'react'

const TargetUrl = props => {
	return (
		<Fragment>
			<div className="input-group my-4">
				<div className="input-group-prepend">
					<button className="btn btn-primary dropdown-toggle" type="button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">{props.isHttp ? 'http://' : 'https://'}</button>
					<div className="dropdown-menu">
						<div className="dropdown-item cursor-pointer" onClick={props.protocolHandler}>{props.isHttp ? 'https://' : 'http://'}</div>
					</div>
				</div>
				<input 
					type="text" 
					className="form-control" 
					placeholder="Введите ссылку на целевую страницу" 
					aria-label="Text input with dropdown button" 
					onChange={event => props.targetUrlHandler(event.target.value)}
					/>
			</div>
		</Fragment>

	)
}

export default TargetUrl