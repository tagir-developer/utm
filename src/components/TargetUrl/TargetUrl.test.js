import React from 'react'
import TargetUrl from './TargetUrl'

describe('<TargetUrl />', () => {

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<TargetUrl />)
	})

	describe('Component render', () => {
	
		it('Should TargetUrl component render', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})
	
		it('Should set right protocol if isHttp: true', () => {
			wrapper.setProps({
				isHttp: true
			})
			expect(wrapper.find('div.dropdown-item.cursor-pointer').text()).toEqual('https://')
		})
	
		it('Should set right protocol if isHttp: false', () => {
			wrapper.setProps({
				isHttp: false
			})
			expect(wrapper.find('div.dropdown-item.cursor-pointer').text()).toEqual('http://')
		})

	})

	describe('Component handlers testing', () => {

		
		it('Should call protocolHandler', () => {
			const clickHandler = jest.fn()
			wrapper.setProps({
				protocolHandler: clickHandler
			})
			wrapper.find('div.dropdown-item.cursor-pointer').simulate('click')
			expect(clickHandler).toHaveBeenCalled()
		})

		it('Should call targetUrlHandler', () => {
			const valueChangeHandler = jest.fn()
			wrapper.setProps({
				targetUrlHandler: valueChangeHandler
			})
			wrapper.find('input.form-control').simulate('change', {target: {value: 'testing'}})
			expect(valueChangeHandler).toHaveBeenCalledWith('testing')
		})

	})

	

})