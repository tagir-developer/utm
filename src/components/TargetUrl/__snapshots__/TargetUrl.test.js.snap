// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TargetUrl /> Component render Should TargetUrl component render 1`] = `
<Fragment>
  <div
    className="input-group my-4"
  >
    <div
      className="input-group-prepend"
    >
      <button
        aria-expanded="false"
        aria-haspopup="true"
        className="btn btn-primary dropdown-toggle"
        data-toggle="dropdown"
        type="button"
      >
        https://
      </button>
      <div
        className="dropdown-menu"
      >
        <div
          className="dropdown-item cursor-pointer"
        >
          http://
        </div>
      </div>
    </div>
    <input
      aria-label="Text input with dropdown button"
      className="form-control"
      onChange={[Function]}
      placeholder="Введите ссылку на целевую страницу"
      type="text"
    />
  </div>
</Fragment>
`;
